src/software/SfM/main_openMVG2NVM.cpp

```diff
@@ -0,0 +1,273 @@
+                    const std::string filename )
+{
+  const string sOutViewsDirectory2 = "views" ;
+  const string sOutViewsDirectory = stlplus::folder_append_separator( sOutDirectory ) + "views";
+  if ( !stlplus::folder_exists( sOutViewsDirectory ) )
+  {
+    cout << "\033[1;31mCreating directory:  " << sOutViewsDirectory << "\033[0m\n";
+    stlplus::folder_create( sOutViewsDirectory );
+  }
+
+  // Header
+  std::ofstream file( filename ) ;
+
+  if( ! file )
+  {
+    std::cerr << "Cannot write file" << filename << std::endl ;
+    return false ;
+  }
+  file << "NVM_V3" << std::endl ;
+
+  // we reindex the poses to ensure a contiguous pose list.
+  Hash_Map<IndexT, IndexT> map_viewIdToContiguous;
+  int nb_cam = 0 ;
+  for( Views::const_iterator iter = sfm_data.GetViews().begin();
+       iter != sfm_data.GetViews().end(); ++iter )
+  {
+    const View * view = iter->second.get();
+    if ( !sfm_data.IsPoseAndIntrinsicDefined( view ) )
+    {
+      continue;
+    }
+
+    nb_cam ++ ;
+    map_viewIdToContiguous.insert( std::make_pair( view->id_view, map_viewIdToContiguous.size() ) );
+  }
+
+  // Number of cameras
+  // For each camera : File_name Focal Qw Qx Qy Qz Cx Cy Cz D0 D1 0
+
+  file << nb_cam << std::endl ;
+  std::string sOutViewIteratorDirectory;
+  std::string sOutViewIteratorDirectory2 ;
+  Image<RGBColor> image, image_ud ;
+  C_Progress_display my_progress_bar( sfm_data.GetViews().size() );
+  for( Views::const_iterator iter = sfm_data.GetViews().begin();
+       iter != sfm_data.GetViews().end(); ++iter , ++my_progress_bar )
+  {
+    const View * view = iter->second.get();
+
+    if ( !sfm_data.IsPoseAndIntrinsicDefined( view ) )
+    {
+      continue;
+    }
+
+    std::ostringstream padding;
+    padding << std::setw( 4 ) << std::setfill( '0' ) << view->id_view;
+    sOutViewIteratorDirectory = stlplus::folder_append_separator( sOutViewsDirectory ) + "view_" + padding.str() ;
+    sOutViewIteratorDirectory2 = stlplus::folder_append_separator( sOutViewsDirectory2 ) + "view_" + padding.str() ;
+
+    const std::string srcImage = stlplus::create_filespec( sfm_data.s_root_path, view->s_Img_path );
+    const std::string dstImage =
+      stlplus::create_filespec( stlplus::folder_append_separator( sOutViewIteratorDirectory ), "undistorted", "png" );
+    const std::string dstImage2 =
+      stlplus::create_filespec( stlplus::folder_append_separator( sOutViewIteratorDirectory2 ), "undistorted", "png" );
+
+    // Create output dir if not present
+    if ( !stlplus::folder_exists( sOutViewIteratorDirectory ) )
+    {
+      stlplus::folder_create( sOutViewIteratorDirectory );
+    }
+
+
+
+    Intrinsics::const_iterator iterIntrinsic = sfm_data.GetIntrinsics().find( view->id_intrinsic );
+    const IntrinsicBase * cam = iterIntrinsic->second.get();
+    const Pinhole_Intrinsic * pinhole_cam = static_cast<const Pinhole_Intrinsic *>( cam );
+    const double flen = pinhole_cam->focal() ;
```

Comment:
You're right for the moment all camera model inherit from pinhole_intrinsic so there is not prb, yet.